### ISSUE TITLE
Running `cargo make audit` in a workspace does not produce useful behaviour

### ISSUE THREAD

**Issue by azriel91** [sagiegurari/cargo-make#74]
#### Problem Description

Running `cargo make audit` in a workspace does not produce useful behaviour &mdash;
`cargo audit` is attempted to be run inside each workspace member, but fails on the first because `Cargo.lock` is not found.

```
[cargo-make] INFO - Execute Command: "cargo" "audit"
error: Couldn't find 'Cargo.lock'!

Run "cargo build" to generate lockfile before running audit
```

---

The following table shows what I kind of expect when running `cargo make audit`:

| Working directory | Expected Behaviour |
| --- | --- |
| workspace root | `cargo audit` is run once over the entire workspace |
| member crate | `cargo audit` is either not run at all, or is run, passing the workspace `Cargo.lock` as an argument |

I was attempting to fix this in https://github.com/azriel91/cargo-make/tree/feature/cargo-audit-for-workspace, but then discovered the `"workspace"` task is generated in code. That's where I stopped :grimacing:

Oh ya, should the `audit` task also have `install_crate = "cargo-audit"`? I didn't check if it automatically does

**Comment by sagiegurari** [comment]
Thanks I'll fix that.
For now you can workaround it by overriding the task and just adding

```toml 
[tasks.audit]
cwd = ".."
```

it **_should_** work (I hope)

**Comment by sagiegurari** [comment]
cargo-make 0.10.3 has been released with a fix which does the following:
Only if Cargo.lock exists in working directory will the audit be invoked. so for members in workspace this will be skipped. You will have to run in the workspace with the no workspace flag to run this for workspace level audit.

As for the install_crate, there is no need to add install_crate in most cases.
cargo-make already knows how to do that automatically so that was not an issue.
Forgot to mention that in my original reply.

**Comment by sagiegurari** [comment]
@azriel91 can you please confirm the fixes are working good for you?

**Comment by azriel91** [comment]
Hey, sorry for not testing sooner; on `0.10.3`, `cargo make --no-workspace audit` is nearly a no-op on the workspace root:

```
$ cargo make --no-workspace audit           
[cargo-make] INFO - cargo-make 0.10.3
[cargo-make] INFO - Using Build File: Makefile.toml
[cargo-make] INFO - Task: audit
[cargo-make] INFO - Setting Up Env.
[cargo-make] INFO - Running Task: init
[cargo-make] INFO - Running Task: audit
[cargo-make] INFO - Running Task: end
[cargo-make] INFO - Build Done  in 0 seconds.
```

I haven't checked the code why, but it may be because this is a virtual workspace:

```toml
# Cargo.toml
[workspace]
members = ["app/*", "crate/*"]
```

I've confirmed it doesn't run `cargo audit` a member crate if you happen to run it in there:

```
$ cargo make audit               
[cargo-make] INFO - cargo-make 0.10.3
[cargo-make] INFO - Using Build File: Makefile.toml
[cargo-make] INFO - Task: audit
[cargo-make] INFO - External file not found, skipping.
[cargo-make] INFO - Setting Up Env.
[cargo-make] INFO - Running Task: init
[cargo-make] INFO - Running Task: audit
[cargo-make] INFO - Running Task: end
[cargo-make] INFO - Build Done  in 0 seconds.
```

**Comment by sagiegurari** [comment]
Ok, I'll run few more manual tests on workspaces as well tomorrow.
On members, it should do nothing as you saw.

**Comment by sagiegurari** [comment]
I found the issue and will push a fix later today/tomorrow.

**Comment by sagiegurari** [comment]
fixed in 0.10.5, the issue was that I also checked dependencies exist in the cargo.toml but in the workspace level, not always the dependencies are defined there, so I now set the dependencies env var to true not just if dependencies exist in the toml file, but also if we are working in a workspace level cargo.toml file.
Hopefully that should resolve it.

**Comment by azriel91** [comment]
Yeaps it works now, thank you!
